Swaps: Alpha 2

wachunei opened the pull request.

wachunei committed:
Swaps: Add priceSlippage alert (#2151)

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Esteban Mino <efmino@uc.cl>
Co-authored-by: tlip <flamingYawn@users.noreply.github.com>
Co-authored-by: tlip <11626601+tlip@users.noreply.github.com>

wachunei committed:
Swaps: Add verify token message and link (#2132)

wachunei committed:
Swaps: Add action button and SwapsLiveness component (#2117)

Co-authored-by: Ricky Miller <ricky.miller@gmail.com>

wachunei committed:
Swaps: Asset selector ordering (#2121)

Co-authored-by: Ricky Miller <ricky.miller@gmail.com>

wachunei committed:
Swaps: Loading Animation 🦊 (#2105)

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Esteban Mino <efmino@uc.cl>
Co-authored-by: tlip <flamingYawn@users.noreply.github.com>
Co-authored-by: tlip <11626601+tlip@users.noreply.github.com>

wachunei committed:
Swaps: Add fiat amount in preferred currency (#2135)

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Erik Marks <25517051+rekmarks@users.noreply.github.com>
Co-authored-by: Erik Marks <rekmarks@protonmail.com>
Co-authored-by: Esteban Mino <efmino@uc.cl>
Co-authored-by: tlip <flamingYawn@users.noreply.github.com>
Co-authored-by: tlip <11626601+tlip@users.noreply.github.com>
Co-authored-by: Andre Pimenta <andrepimenta7@gmail.com>

estebanmino committed:
REVERTTHIS

estebanmino committed:
REVERTTHISAGAIN

estebanmino committed:
ifactive

estebanmino committed:
active

estebanmino committed:
Merge branch 'engine-swaps-flag' into feature/swaps-alpha-2

wachunei committed:
Swaps: Fix amount and use max in amount view (#2159)

estebanmino commented on ios/MetaMask.xcodeproj/project.pbxproj:
```diff
@@ -847,7 +847,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Manual;
-				CURRENT_PROJECT_VERSION = 564;
+				CURRENT_PROJECT_VERSION = 565;
```
```suggestion
				CURRENT_PROJECT_VERSION = 566;
```

estebanmino commented on ios/MetaMask.xcodeproj/project.pbxproj:
```diff
@@ -910,7 +910,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Distribution";
 				CODE_SIGN_STYLE = Manual;
-				CURRENT_PROJECT_VERSION = 564;
+				CURRENT_PROJECT_VERSION = 565;
```
```suggestion
				CURRENT_PROJECT_VERSION = 566;
```

estebanmino committed:
Update ios/MetaMask.xcodeproj/project.pbxproj

estebanmino committed:
Update ios/MetaMask.xcodeproj/project.pbxproj

wachunei committed:
Fix topAssets bug when is not in token list

wachunei committed:
Fix use max balance

token balance are already a BN instance

wachunei committed:
Swaps: Add wallet action buttons (#2160)

estebanmino committed:
570

estebanmino committed:
constants

estebanmino committed:
merge

wachunei committed:
Fix bignumber balance

estebanmino committed:
1.0.11

wachunei committed:
Fix null balanceAsUnits bug

wachunei committed:
Bump ios version

wachunei committed:
Merge branch 'feature/swaps-alpha-2' of github.com:MetaMask/metamask-mobile into feature/swaps-alpha-2

wachunei committed:
Fix balance bug and bump to 573

wachunei committed:
Add hitSlop area to slippage button

wachunei committed:
Swaps: keypad long press delete (#2182)

wachunei committed:
Swaps: Add keyboard shake (#2181)

wachunei committed:
Use isZero in for balance

wachunei committed:
Swaps: Quotes details (#2173)

estebanmino committed:
fix

estebanmino committed:
chinid

estebanmino committed:
engine

estebanmino committed:
removerefresh

estebanmino committed:
test

ejwessel committed:
Swaps: Swap Metrics (#2150)

* Squashed commit of the following:

commit d552524864c1a243bd7b3ad727582cc6f9e9d751
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Tue Nov 17 14:05:22 2020 -0300

    Remove unnecessary styling

commit 24b545666204f713ec36dc14cb0a8cf00574a413
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Tue Nov 17 13:13:16 2020 -0300

    Update tokens dependency for effect

commit 1662700e4ac435a5ccdb1abb6d7d9d502c049d30
Author: Esteban Mino <efmino@uc.cl>
Date:   Tue Nov 17 12:50:48 2020 -0300

    bumpcontrollers

commit 94a14dd5cdb835307081f0ec6990d9663fd42993
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Mon Nov 16 13:52:01 2020 -0300

    Update main navigation snapshot

commit f9ba8af6185514da7eb5c5fa211c880981672d7f
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Mon Nov 16 13:43:14 2020 -0300

    Add TokenSelectButton label prop to test

commit 637452fe40f884ea50a9571b701930d1c135d22b
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Mon Nov 16 13:34:37 2020 -0300

    Add i18n to swaps amount view

commit 6dfd9a4a102e7c4e690e97b208b23f30ed89df42
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Fri Nov 13 15:52:21 2020 -0300

    Fix TokenIcon border radius

commit 54c17c2a76723b0b0ae97ca6271b0069c3fa608a
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Fri Nov 13 13:30:16 2020 -0300

    Remove unused key in token search for swaps

commit 4bb4bcb3b42dcf2b43fee9b0776e49d6e9baf9f1
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Thu Nov 12 23:30:58 2020 -0300

    Add asset balance in swaps amount view

commit b262b4e3b3bac640872fd96693be3abd0b97e4a4
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Thu Nov 12 22:48:36 2020 -0300

    Add token balances in token search modal list

commit 6ca34c9db64f3b2504350c7c03da338fc5665b34
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Thu Nov 12 14:04:33 2020 -0300

    Add invalid decimal warning

commit 539d1e775b0d5be4bb1019bf8cc27061c416bd18
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Wed Nov 11 23:11:26 2020 -0300

    Add empty list message and clear search on modal hide

commit 65a31a6c6f73355beeb4c310de9c955969763b44
Author: Esteban Mino <efmino@uc.cl>
Date:   Wed Nov 11 22:00:20 2020 -0300

    bumpcontrollers

commit 36a22894e0620114c92d988396592c8db85a1fd1
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Wed Nov 11 19:18:40 2020 -0300

    Update Receiver Modal snapshot, it uses ModalHandler now

commit c69c414e30d6c5b0cc8e9c1edf3778ee50bcb57e
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Wed Nov 11 19:13:31 2020 -0300

    Add tokens from SwapsController

commit 4f4e87f8ecd03bcea93c668a3142cc66e667dd65
Author: Esteban Mino <efmino@uc.cl>
Date:   Tue Nov 10 21:57:42 2020 -0300

    inception

commit 23673938ffe221f08c1438cf8c228b540053bf20
Author: Esteban Mino <efmino@uc.cl>
Date:   Tue Nov 10 19:09:15 2020 -0300

    package

commit 707b491d222d5d3176274b99c78e65434dce92b5
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Wed Nov 11 17:49:41 2020 -0300

    Add Initial Token Selector with dummy tokens

commit 8644be0f3bbd7da00a1d3912273042a4da709984
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Tue Nov 10 12:40:34 2020 -0300

    Switch Swaps feature flag to false

commit 0696f1071f54dab08fc419a4b832ee427a9f40b1
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Mon Nov 9 20:40:44 2020 -0300

    Add Keypad rules and swap token button

commit 31732901d98ff5de1eaed10ea4ff28a7ad84daf5
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Fri Nov 6 17:08:56 2020 -0300

    Update snapshot for main nav

commit 5f031985422e916c08a7eaa9b344562dfa8d4b18
Author: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Date:   Fri Nov 6 12:41:01 2020 -0300

    Add SWAPS.ACTIVE flag

* Disable flag

* Remove console.log

* Fix linting

* bump and getch quotes

* bump

* bump

* fetchingcorrectly

* fetchQuotesfetchQuotes

* Use SwapsController state

* Add initial generic error view

* bumpcontroller

* Add swaps app utils

* Add ratio and values from top quote

* bump

* bump

* bump

* effect

* Add gas price

* Remove console.log

* Add quotes modal and i18n

* tradefees

* bump

* tradefees

* orderedtradesfees

* bump

* bumpcontrolker

* snaps

* walletaddress

* bump

* bump

* quotevalues

* Order quotes by overall quote value

* Add enable asset text row

* bump

* bump

* 500

* Swaps amount view: use max for tokens (#2052)

* Add slider button component (#2046)

* Add slider buttons and useBalance

useBalance should be used to show balance alerts

* detecting

* bump

* REVERTTHISVALIDATION

* Add Alert component (#2061)

* Add balance alert to quotes view

* fixsendto

* networkfix

* Fix use max handler usecallback deps

* Add Slippage Slider Component (#2068)

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>

* controllers

* circle

* circlerevert

* 9

* bumpcontrollers

* CURRENT_PROJECT_VERSION

* Refactor slider button

* Add visual feedback

- Nav title
- Spacing
- Text color and size

* Adjust token symbol text colors

* Update ios/MetaMask.xcodeproj/project.pbxproj

* Update ios/MetaMask.xcodeproj/project.pbxproj

* Fix SliderButton infinite effect

* Update ios/MetaMask.xcodeproj/project.pbxproj

* Update ios/MetaMask.xcodeproj/project.pbxproj

* gasprice

* 561

* Make onComplete optional

* averagegas

* 42

* Add touchable ratio switcher

* Add default 3% slippage constant

* Add fee variable to fee modal

* Add hitSlop area to modal close buttons

* Add quotes overview string

* Add error views styling

* Metaswap Metrics

* Add Swap Actions

* move swap actions to names; actions are wip

* add analytics event opt

* merge

* networkupdated

* publish swap open metric

* logstomixpanel

* moredev

* remove depdendency on initialSource

* add back depdendency on initialSource to prevent multi publish

* add quote view metrics

* checkpoint for quotes received

* quotes received metrics

* quotes timeout metric

* remove console log

* add metric to quotes modal opening, no data sent

* publish quotes modal with data

* swaps started metrics

* update timed out and no quotes available metrics

* fix broken styling

* remove param

* handle custom slippage detection

* update swaps opened view metric and custom slippage

* update request type data point in metrics

* datapoint all available quotes

* other quote selected data point

* available quotes and remove console log

* available quotes metric

* additional data point

* move quotes received metric

* added best quote source for quotes received

* max fetch time for all quotes

* request type order for modal view

* add InteractionManager

* remove metrics from quotes modal

* remove commented out lines

* add param to nav bar

* publish cancel quote metrics on dismissal of quote request

* publish response time for cancellation

* run after interactions

* use callbacks

* detect when user has backed out before a trade has been selected

* handle navigation param

* implement feedback

* fix inconsistencies in metrics

* comments for react-hooks/exhaustive-dep disabled

* add state for request and received

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Esteban Mino <efmino@uc.cl>
Co-authored-by: tlip <flamingYawn@users.noreply.github.com>
Co-authored-by: tlip <11626601+tlip@users.noreply.github.com>

estebanmino committed:
Feature/swaps gas and approval edition (#2163)

* renamevars

* renderingcurstomgas

* selectgas

* txwithcustomprice

* onEditMaxGasCancel

* gaslimittotrade

* flag

* usecorrectgas

* logs

* gasnaming

* customtx

* wip

* editapproval

* wip

* gas?

* gas

* uiworking

* modifytxdata

* component

* gasworking

* gasselector

* approvaldata

* navbar

* working

* works

* snaps

* callbacks

* comment

* close

* close

wachunei committed:
Swaps: Flip selected tokens in amount view (#2190)

estebanmino committed:
fix

wachunei committed:
Swaps: Onboarding (#2200)

estebanmino commented:
reminder to turn the feature off when we merge this

wachunei committed:
Swaps: Action Alerts for tokens and slippage (#2214)

wachunei committed:
Swaps: Add liveness check and remove from drawer (#2215)

estebanmino committed:
Swaps: Quotes timeout (#2216)

* isinfetchincomponent

* bump

estebanmino committed:
merge

estebanmino committed:
Merge branch 'improvement/handle-provider-updates' into feature/swaps-alpha-2

estebanmino committed:
booleanfix

estebanmino committed:
574

wachunei committed:
Fix onboarding image size

wachunei committed:
Swaps: Add buy button (#2192)

wachunei committed:
Make animation faster

estebanmino committed:
Swaps: Eth balance validation (#2218)

* hasEnoughEthBalance

* eth

* works

* works

* sameusememo

* tests

* small

* swapsutils

* swapsutils

* logger

estebanmino committed:
575

wachunei committed:
Swaps: fix Use Max by adding `fromTokenMinimalUnitString` (#2224)

wachunei committed:
Add ETH to topAssets

wachunei committed:
Swaps: Add terms of service (#2229)

estebanmino committed:
More gas stuff (#2235)

* currency and unable to swap

* contractmetadata

* remotesvg

* gasvalidationagainstmaxgasfee

* controllers

* snaps

wachunei committed:
Swaps: Add information modals (#2241)

wachunei committed:
Check isZero function exists

estebanmino committed:
bugfix/edit permission validations (#2239)

* permission

* strings

estebanmino committed:
Swaps/transaction labels (#2244)

* Remove best deals badge from WYRE transfers. (#2228)

Co-authored-by: Shivam Chaturvedi <corsairshivam@gmail.com>

* Handle `balanceError` case (#2148)

* Handle `balanceError` case

* Adjust error case

* Update handling balanceError

* Remove etherscan link

* Add metamask-controllers from disk

* Add translation strings

* Remove etherscanUrl

* Revert "Add metamask-controllers from disk"

This reverts commit 75ac0c68b99e5bf2ff190c20cccad4df74a8806f.

* Only use this style for error case

* contract metadata images bump (#2233)

* contractmetadata

* snaps

* remotesvg

* Feature/warn when replacing (#2176)

* mock login changes

* mock warning modal

* mock delete modal

* mock delete warning and submitDelete

* Add isTextDelete

* Add autoFocus

* clean up

* "delete"

* undo design changes

* Add en translations

* Update unit tests

* fix

* Make this async await and catch and log failure

* Remove 12-word

* Adjust lineHeight

* fix typo

* Add logic to show delete notification

* Hide back button

* Disable hardware back button press

* Add notification

* Add closeButtonDisabled functionality

* Update styles

* Hide back button better on Congratulations screen

* Update translation

* Add breakpoint for small devices

* add missing "a"

* fix warning modal

* Clean up BaseNotification component

* Remove added styles

* Remove eslint-disable-next-line

* Use InteractionManager

* Remove empty onHide

* update empty callback

* Cleanup

* Add flexDirection: 'row'

* Move styles to WarningExistingUserModal

* Dismiss keyboard

* Add strings to locales

* Simplify

* Add loading screen for delete

* Actually delete wallet

* Fix modal tap area

Co-authored-by: ricky <ricky@macbook.lan>

* v1.0.11 (#2238)

* version

* bump

* Update CHANGELOG.md

* update

* Update CHANGELOG

* Remove disclosure PR

* Fix circleci apk (#2240)

* resouceclass

* revert

* Revert "revert"

This reverts commit 60df2aa518bbe2d3bcf60788594ae089c6c78e67.

* changelog

* gaba

* swaps-tx

* MM_FOX_CODE

* swaptxs

* works

* fix

* TransactionController from controllers (#2243)

* TransactionController from controllers

* query

* changelog

* tests

* 577

* 577

* details

* transactionType

* token

* byaction

* allworking

* ceanup

* locales

* utils

* check

* edgecase

* test

* tests

* approvals

* filter

* more

Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Shivam Chaturvedi <corsairshivam@gmail.com>
Co-authored-by: ricky <ricky.miller@gmail.com>
Co-authored-by: ricky <ricky@macbook.lan>

estebanmino committed:
578

wachunei committed:
Merge branch 'develop' into feature/swaps-alpha-2

wachunei committed:
Swaps: Add mobile_active flag for liveness (#2248)

* Add mobile_active flag for liveness

* bump

Co-authored-by: Esteban Mino <efmino@uc.cl>

estebanmino committed:
Notifications refactor for swaps (#2258)

* functionalcomponent

* more

* txnnotification

* tnotification

* simplenotigication

* contructnot

* queue

* autodismiss

* byid

* HIDE_CURRENT_NOTIFICATION

* simplify

* simple

* listeners

* REMOVE_NOTIFICATION_BY_ID

* notificationlabel

* notificationlabel

* betterlocales

* gaslimit

* press

* speedcancel

* gasLimitWithMultiplier

* comments

* useCallback

* SimpleNotification

estebanmino committed:
bump1.0.12

estebanmino committed:
580

wachunei committed:
Swaps: Custom gas/approval analytics (#2261)

wachunei committed:
Swaps: Notifications reducer refactor (#2263)

estebanmino committed:
swaps: Several bugfixes (#2266)

* tokenbalancesinterval

* label-fixes

* txdeatilspadding

* approvaledit

* TokenImage

* test

* test

wachunei committed:
Fix misaligned buy more eth on alert (#2280)

wachunei committed:
Fix gas information alignment on small screens (#2282)

wachunei committed:
Fix edit limit out of box (#2279)

estebanmino committed:
swaps: Missing analytics (#2270)

* confirmandfailed

* usesymbol

* _trackEvent

* trackSwaps

* swapsTransactions

* ggasused

* getSwapTransactionAnalyticsInformation

* working?

* works

* ratesworking

* replacedestinationAmount

* betterwording

* gettimewithblocks

* afterinteractions

* logger

* analyticsParams

* tokensReceived

* tokensReceived

wachunei committed:
Transactions view refactor (#2291)

Co-authored-by: andrepimenta <andrepimenta7@gmail.com>

wachunei committed:
Fix custom gas alert margin (#2290)

wachunei committed:
Use token format in analytics (#2293)

estebanmino committed:
swaps: fixes (#2292)

* updateSwapsTransactions

* 2273

* 2285

* 2288

* snaps

estebanmino committed:
581

estebanmino committed:
swaps: fix retry fetch quotes (#2294)

estebanmino committed:
Swaps: track polling errors (#2295)

* logger

* bump

estebanmino committed:
582

wachunei committed:
Swaps: Remove repeated useEffect in loading animation (#2296)

estebanmino committed:
583

estebanmino committed:
swaps: Gas limit with multiplier (#2298)

estebanmino committed:
584

wachunei committed:
Remove settimeout 0 on complete (#2300)

estebanmino committed:
585

estebanmino committed:
Transactions destination amount default (#2305)

* changeifdefined

* decimalPlaces

* saveamountsasminimalunit

* bump

* test

* updateswapsbeforeanalytics

* test

estebanmino committed:
586

estebanmino committed:
swaps approval label (#2310)

* fix

* fix

wachunei committed:
Add props to hide selectors from gas edit modal (#2311)

ejwessel committed:
Switch Naming of Metrics (#2312)

* Switch Naming of Metrics

* REVERTTHIS

* Revert "REVERTTHIS"

This reverts commit ed7c90dafba37ef3336c834ebd67e283c178150a.

* rmcomment

* SWAP_TRACKING_FAILED

Co-authored-by: Esteban Mino <efmino@uc.cl>

estebanmino committed:
mergedev

estebanmino committed:
587

estebanmino committed:
flags

ibrahimtaveras00 reviewed (approved):
Alpha QA Passed ✅ 
ready for beta
